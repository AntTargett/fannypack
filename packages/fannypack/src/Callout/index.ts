import * as styles from './styles';
import {
  Callout as _Callout,
  CalloutHeader,
  CalloutTitle,
  CalloutContent,
  CalloutFooter,
  CalloutIcon
} from './Callout';

export * from './Callout';
export const Callout = Object.assign(_Callout, {
  Header: CalloutHeader,
  Title: CalloutTitle,
  Content: CalloutContent,
  Footer: CalloutFooter,
  Icon: CalloutIcon
});
export { styles as CalloutStyles };
