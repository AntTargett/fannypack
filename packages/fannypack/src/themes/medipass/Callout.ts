import { css } from '../../styled';

export default {
  base: css`
    border-radius: 0px;
  `
};
