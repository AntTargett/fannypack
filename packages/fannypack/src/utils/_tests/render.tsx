import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
