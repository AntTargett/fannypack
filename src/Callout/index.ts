import Callout from './Callout';
import * as calloutStyled from './styled';

export * from './Callout';
export * from './CalloutClose';
export { calloutStyled };
export default Callout;
