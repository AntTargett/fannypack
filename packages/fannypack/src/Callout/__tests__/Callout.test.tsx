import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Callout } from '../Callout';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Callout>Hello world</Callout>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Callout color="primary">Hello world</Callout>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Callout use="div">Hello world</Callout>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Callout props', () => {
      const { result } = renderHook(() => Callout.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Callout>{CalloutProps => <div {...CalloutProps}>Hello world</div>}</Callout>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Callout.base should render correctly', () => {
    const { container } = render(<Callout>hello world</Callout>, {
      // @ts-ignore
      theme: { Callout: { base: { backgroundColor: 'red' } } }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Callout>hello world</Callout>, {
      // @ts-ignore
      theme: { Callout: { defaultProps: { className: 'test', color: 'primary' } } }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
