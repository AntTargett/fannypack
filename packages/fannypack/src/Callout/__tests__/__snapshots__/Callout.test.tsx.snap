// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-19 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-17 {
  padding: 1.5rem;
}

.emotion-15 {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
}

.emotion-6 {
  margin-right: 1rem;
}

.emotion-6 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-6 .fp-Callout.Content .fp-Icon {
  font-size: 1.25em;
}

.emotion-2 {
  color: #1e67d5 !important;
  font-size: 1.25em !important;
}

.emotion-0 {
  fill: #1e67d5;
  position: relative;
  height: 1em;
  width: 1em;
}

<div
  aria-describedby="calloutDescription"
  class="emotion-16 emotion-17 emotion-1 emotion-19 fp-Box fp-Card fp-Callout"
  type="info"
>
  <div
    class="emotion-1 emotion-15 fp-Box"
  >
    <div
      class="emotion-6 emotion-1 emotion-5 fp-Box fp-Callout.IconWrapper"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        <svg
          aria-hidden="true"
          class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
          role="img"
          viewBox="0 0 512 512"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="emotion-1 emotion-5 fp-Box"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`composition hook should return with Callout props 1`] = `
Object {
  "aria-describedby": undefined,
  "aria-labelledby": undefined,
  "children": <Context.Provider
    value={
      Object {
        "descriptionId": "",
        "titleId": "",
        "type": "info",
      }
    }
  >
    <ForwardRef
      display="flex"
    >
      <React.Fragment>
        <ForwardRef />
        <ForwardRef>
          <ForwardRef />
        </ForwardRef>
      </React.Fragment>
    </ForwardRef>
  </Context.Provider>,
  "className": "css-1fz0eok css-gziovy css-1pwrqjj css-ghetqn fp-Box fp-Card fp-Callout",
  "type": "info",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-3 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  padding: 1.5rem;
}

<div
  aria-describedby="calloutDescription"
  class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Card fp-Callout"
  type="info"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-17 {
  padding: 1.5rem;
}

.emotion-15 {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
}

.emotion-6 {
  margin-right: 1rem;
}

.emotion-6 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-6 .fp-Callout.Content .fp-Icon {
  font-size: 1.25em;
}

.emotion-2 {
  color: #1e67d5 !important;
  font-size: 1.25em !important;
}

.emotion-0 {
  fill: #1e67d5;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-19 {
  color: #3722ed !important;
}

.emotion-19 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

<div
  aria-describedby="calloutDescription"
  class="emotion-16 emotion-17 emotion-1 emotion-19 test fp-Box fp-Card fp-Callout"
  type="info"
>
  <div
    class="emotion-1 emotion-15 fp-Box"
  >
    <div
      class="emotion-6 emotion-1 emotion-5 fp-Box fp-Callout.IconWrapper"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        <svg
          aria-hidden="true"
          class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
          role="img"
          viewBox="0 0 512 512"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="emotion-1 emotion-5 fp-Box"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-19 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-17 {
  padding: 1.5rem;
}

.emotion-15 {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
}

.emotion-6 {
  margin-right: 1rem;
}

.emotion-6 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-6 .fp-Callout.Content .fp-Icon {
  font-size: 1.25em;
}

.emotion-2 {
  color: #1e67d5 !important;
  font-size: 1.25em !important;
}

.emotion-0 {
  fill: #1e67d5;
  position: relative;
  height: 1em;
  width: 1em;
}

<div
  aria-describedby="calloutDescription"
  class="emotion-16 emotion-17 emotion-1 emotion-19 fp-Box fp-Card fp-Callout"
  type="info"
>
  <div
    class="emotion-1 emotion-15 fp-Box"
  >
    <div
      class="emotion-6 emotion-1 emotion-5 fp-Box fp-Callout.IconWrapper"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        <svg
          aria-hidden="true"
          class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
          role="img"
          viewBox="0 0 512 512"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="emotion-1 emotion-5 fp-Box"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-17 {
  padding: 1.5rem;
}

.emotion-15 {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
}

.emotion-6 {
  margin-right: 1rem;
}

.emotion-6 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-6 .fp-Callout.Content .fp-Icon {
  font-size: 1.25em;
}

.emotion-2 {
  color: #1e67d5 !important;
  font-size: 1.25em !important;
}

.emotion-0 {
  fill: #1e67d5;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-19 {
  color: #3722ed !important;
}

.emotion-19 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

<div
  aria-describedby="calloutDescription"
  class="emotion-16 emotion-17 emotion-1 emotion-19 fp-Box fp-Card fp-Callout"
  type="info"
>
  <div
    class="emotion-1 emotion-15 fp-Box"
  >
    <div
      class="emotion-6 emotion-1 emotion-5 fp-Box fp-Callout.IconWrapper"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        <svg
          aria-hidden="true"
          class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
          role="img"
          viewBox="0 0 512 512"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="emotion-1 emotion-5 fp-Box"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`theming Callout.base should render correctly 1`] = `
.emotion-19 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-17 {
  padding: 1.5rem;
}

.emotion-15 {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
}

.emotion-6 {
  margin-right: 1rem;
}

.emotion-6 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-6 .fp-Callout.Content .fp-Icon {
  font-size: 1.25em;
}

.emotion-2 {
  color: #1e67d5 !important;
  font-size: 1.25em !important;
}

.emotion-0 {
  fill: #1e67d5;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-16 {
  background-color: red;
}

<div
  aria-describedby="calloutDescription"
  class="emotion-16 emotion-17 emotion-1 emotion-19 fp-Box fp-Card fp-Callout"
  type="info"
>
  <div
    class="emotion-1 emotion-15 fp-Box"
  >
    <div
      class="emotion-6 emotion-1 emotion-5 fp-Box fp-Callout.IconWrapper"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        <svg
          aria-hidden="true"
          class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
          role="img"
          viewBox="0 0 512 512"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="emotion-1 emotion-5 fp-Box"
    >
      <div
        class="emotion-3 emotion-1 emotion-5 fp-Box fp-Callout.Content"
        id="calloutDescription"
      >
        hello world
      </div>
    </div>
  </div>
</div>
`;
